src/app/api/tasks/tasks.logger.ts

```diff
@@ -52,6 +54,28 @@ export class TasksActivityLogger extends BaseService {
         setUpdate()
       }
     }
+
+    if (Array.isArray(this.task.viewers) && Array.isArray(prevTask.viewers)) {
+      const currentViewers = (this.task.viewers as Viewers) || []
+      const prevViewers = (prevTask.viewers as Viewers) || []
+
+      if (
+        (!!currentViewers.length || !!prevViewers.length) &&
+        (currentViewers[0]?.clientId !== prevViewers[0]?.clientId ||
+          currentViewers[0]?.companyId !== prevViewers[0]?.companyId)
```

Comment:
```suggestion
        (currentViewers[0].clientId !== prevViewers[0].clientId ||
          currentViewers[0].companyId !== prevViewers[0].companyId)
```

I think TS will accept this as well since we did a !!currentViewers.length & !!prevViewers.length check. Not sure though